Mathlib/Data/List/Perm.lean

```diff
@@ -788,18 +793,19 @@ theorem length_permutations'Aux (s : List α) (x : α) :
 #align list.length_permutations'_aux List.length_permutations'Aux
 
 @[simp]
-theorem permutations'Aux_nthLe_zero (s : List α) (x : α)
+theorem permutations'Aux_get_zero (s : List α) (x : α)
     (hn : 0 < length (permutations'Aux x s) := (by simp)) :
-    (permutations'Aux x s).nthLe 0 hn = x :: s :=
-  nthLe_permutations'Aux _ _ _ _
-#align list.permutations'_aux_nth_le_zero List.permutations'Aux_nthLe_zero
+    (permutations'Aux x s).get ⟨0, hn⟩ = x :: s :=
+  get_permutations'Aux _ _ _ _
+#align list.permutations'_aux_nth_le_zero List.permutations'Aux_get_zero
 
+set_option linter.deprecated false in
 theorem injective_permutations'Aux (x : α) : Function.Injective (permutations'Aux x) := by
   intro s t h
   apply insertNth_injective s.length x
   have hl : s.length = t.length := by simpa using congr_arg length h
-  rw [← nthLe_permutations'Aux s x s.length (by simp), ←
-    nthLe_permutations'Aux t x s.length (by simp [hl])]
+  rw [← nthLe_permutations'Aux s x s.length (by simp),
+    ← nthLe_permutations'Aux t x s.length (by simp [hl])]
```

Comment:
Can't you use `get_permutations'Aux` here?

Reply:
I tried; it wasn't quite straightforward. (Be my guest to also try - or we can postpone this.)

Reply:
I just tried again: the `rw`s still fire, but using `get_permutation'Aux` means the last `simp` doesn't close the goal.

The goal before the `simp` changes from
`nthLe (permutations'Aux x s) (length s) ⋯ = nthLe (permutations'Aux x t) (length s) ⋯`
to
`get (permutations'Aux x s) { val := length s, isLt := ⋯ } = get (permutations'Aux x t) { val := length s, isLt := ⋯ }`;
the ... terms are the same, but I don't know why this makes a difference.

Reply:
It's because the index references the length of the list in its type, so you can't "swap out" the list while leaving the index there. Anyway, I just pushed a fix.